kratos/input_output/vtk_output.cpp

```diff
@@ -737,6 +819,45 @@ void VtkOutput::WriteVectorContainerVariable(
 /***********************************************************************************/
 /***********************************************************************************/
 
+template<typename TContainerType, class TVarType>
+void VtkOutput::WriteIntegrationVectorContainerVariable(
+    const TContainerType& rContainer,
+    const Variable<TVarType>& rVariable,
+    std::ofstream& rFileStream) const
+{
+    if (rContainer.size() == 0) {
+        KRATOS_WARNING("VtkOutput") << mrModelPart.GetCommunicator().GetDataCommunicator() << "Empty container!" << std::endl;
+        return;
+    }
+
+    const int res_size = static_cast<int>((rContainer.begin()->GetValue(rVariable)).size());
+
+    rFileStream << rVariable.Name() << " " << res_size << " " << rContainer.size() << "  float\n";
+
+    // Auxiliar values
+    const auto& r_process_info = mrModelPart.GetProcessInfo();
+    auto& r_this_geometry_begin = (rContainer.begin())->GetGeometry();
+    const GeometryData::IntegrationMethod this_integration_method = (rContainer.begin())->GetIntegrationMethod();
+    const auto& r_integration_points = r_this_geometry_begin.IntegrationPoints(this_integration_method);
+    const SizeType integration_points_number = r_integration_points.size();
+
+    TVarType aux_value;
+    for (auto& r_entity : rContainer) { // TODO: GetValueOnIntegrationPoints should be const methods
+        aux_value = rVariable.Zero();
+        std::vector<TVarType> aux_result(integration_points_number);
+        r_entity.GetValueOnIntegrationPoints(rVariable, aux_result, r_process_info);
```

Comment:
why aren't you using `CalculateValuesOnIntegrationPoints`?

Reply:
GetValueOnIntegrationPoints calls CalculateValuesOnIntegrationPoints if the value is not defined on the element (at least we have done that way in the base solid) 

See https://github.com/KratosMultiphysics/Kratos/blob/fd08b0dc4a4d549e8d295bcaffc32a7856d08481/applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp#L1306

Reply:
but `GetValue...` is semi-deprecated and will be removed with #2194 
if we start using it here then it will make the transition more difficult

Reply:
OK, I still not get the reason of the change in #2194